beacon-chain/db/iface/interface.go

```diff
@@ -22,6 +22,8 @@ type ReadOnlyDatabase interface {
 	Block(ctx context.Context, blockRoot [32]byte) (*eth.SignedBeaconBlock, error)
 	Blocks(ctx context.Context, f *filters.QueryFilter) ([]*eth.SignedBeaconBlock, [][32]byte, error)
 	BlockRoots(ctx context.Context, f *filters.QueryFilter) ([][32]byte, error)
+	BlocksBySlot(ctx context.Context, slot uint64) ([]*eth.SignedBeaconBlock, [][32]byte, error)
```

Comment:
```suggestion
	BlocksBySlot(ctx context.Context, slot uint64) ([]*eth.SignedBeaconBlock,  error)
```

I much prefer to not return roots for `BlocksBySlot`, I like single responsibility principle. The issue also suggested `([]*eth.SignedBeaconBlock, bool, error)` which this diverts from. The ability to return if any block exists instead of length check is nice too